### ISSUE TITLE
Build error - WrongConstant - Gradle - React Native build for android

### ISSUE THREAD

**Issue by leonardo-coutinho-dev** [lottie-react-native/lottie-react-native#1150]
I have the following error when trying to build a react native (bare workflow) app for android (cd android && ./gradlew build):

``` node_modules\lottie-react-native\android\src\main\java\com\airbnb\android\react\lottie\LottieAnimationViewPropertyManager.kt ```

**lines 72 & 87 - 2 errors**

```
C:\Projetos\hph-services-app\node_modules\lottie-react-native\android\src\main\java\com\airbnb\android\react\lottie\LottieAnimationViewPropertyManager.kt:72: Error: Must be one of: Typeface.NORMAL, Typeface.BOLD, Typeface.ITALIC, Typeface.BOLD_ITALIC [WrongConstant]
                return typeface?: Typeface.defaultFromStyle(400)
                                                            ~~~
C:\Projetos\hph-services-app\node_modules\lottie-react-native\android\src\main\java\com\airbnb\android\react\lottie\LottieAnimationViewPropertyManager.kt:87: Error: Must be one of: Typeface.NORMAL, Typeface.BOLD, Typeface.ITALIC, Typeface.BOLD_ITALIC [WrongConstant]
                return typeface?: Typeface.defaultFromStyle(400)
                                                            ~~~

   Explanation for issues of type "WrongConstant":
   Ensures that when parameter in a method only allows a specific set of
   constants, calls obey those rules.
```
## Problem:

The **lint error** points to the following line in the **LottieAnimationViewPropertyManager.kt** file:

```
return typeface?: Typeface.defaultFromStyle(400)
```

### The error message states:

```
Error: Must be one of: Typeface.NORMAL, Typeface.BOLD, Typeface.ITALIC, Typeface.BOLD_ITALIC [WrongConstant]
```

I have:

Gradle version: 7.6.2

```
"react-native": "0.72.6",
"lottie-react-native": "^6.4.1"
```

**Comment by eTryp** [comment]
I managed to resolve the compilation issue with a quick fix. It might not be the most elegant solution, but it is effective.
To implement this fix:

1. Open the file at the following path:

> /node_modules/lottie-react-native/android/src/main/java/com/airbnb/android/react/lottie/LottieAnimationViewPropertyManager.kt

2. In the init method of this file, locate and remove the two instances of `Typeface.defaultFromStyle(400).`

**Comment by TheRogue76** [comment]
Hi guys,

Thanks for pointing this out. This is a linting error. I'll open a PR and update it. One question though mostly out of curiosity, is this actually blocking you guys from building the app? If so we can do a patch release. Our CI is running this fine and we haven't had an issue with this till now. I can see the environment you are building on is Windows though.

**Comment by eTryp** [comment]
Hi,

I'm using MacOS, and the issue arises when I execute `./gradlew build` from the terminal. 
Interestingly, I encounter no issues when running `yarn android`. 

As you pointed out, it seems the linter might only be activated through the build command. 
I'm also interested in understanding @leonardo-coutinho-dev's environment / use case.

**Comment by TheRogue76** [comment]
I wrote the PR. Please wait for it to be approved, and we can probably release 6.5.1 afterwards if all is good.

**Comment by matinzd** [comment]
Can you drop your app's `build.gradle` in here? @leonardo-coutinho-dev

**Comment by matinzd** [comment]
Fixed in [v6.5.1](https://github.com/lottie-react-native/lottie-react-native/releases/tag/v6.5.1) 🚀

**Comment by leonardo-coutinho-dev** [comment]
@eTryp 

Hi, I was trying to build the app on Windows, for Android, through the bash terminal (with **./gradlew build** on android folder).

We already have a build for iOs (no issues).

I was in charge today to create the build for Android and run into this issue in the morning.

Here's the ```build.gradle``` @matinzd:

### project-level

```
// Top-level build file where you can add configuration options common to all sub-projects/modules.

buildscript {
    ext {
        buildToolsVersion = "33.0.0"
        minSdkVersion = 21
        compileSdkVersion = 33
        targetSdkVersion = 33

        // We use NDK 23 which has both M1 support and is the side-by-side NDK version from AGP.
        ndkVersion = "23.1.7779620"

        // react-native-leaflet-view - build solution

        kotlinVersion = "1.7.20" // 1 <- here is solution
        RNNKotlinVersion = kotlinVersion // 2 <- here is solution
    }
    repositories {
        google()
        mavenCentral()
    }
    dependencies {
        classpath("com.android.tools.build:gradle")
        classpath("com.facebook.react:react-native-gradle-plugin")

        // react-native-leaflet-view - build solution

        classpath("org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlinVersion")// 3 <- here is solution
    }
}
```

### app-level

```
apply plugin: "com.android.application"
apply plugin: "com.facebook.react"

/**
 * This is the configuration block to customize your React Native Android app.
 * By default you don't need to apply any configuration, just uncomment the lines you need.
 */
react {
    /* Folders */
    //   The root of your project, i.e. where "package.json" lives. Default is '..'
    // root = file("../")
    //   The folder where the react-native NPM package is. Default is ../node_modules/react-native
    // reactNativeDir = file("../node_modules/react-native")
    //   The folder where the react-native Codegen package is. Default is ../node_modules/@react-native/codegen
    // codegenDir = file("../node_modules/@react-native/codegen")
    //   The cli.js file which is the React Native CLI entrypoint. Default is ../node_modules/react-native/cli.js
    // cliFile = file("../node_modules/react-native/cli.js")

    /* Variants */
    //   The list of variants to that are debuggable. For those we're going to
    //   skip the bundling of the JS bundle and the assets. By default is just 'debug'.
    //   If you add flavors like lite, prod, etc. you'll have to list your debuggableVariants.
    // debuggableVariants = ["liteDebug", "prodDebug"]

    /* Bundling */
    //   A list containing the node command and its flags. Default is just 'node'.
    // nodeExecutableAndArgs = ["node"]
    //
    //   The command to run when bundling. By default is 'bundle'
    // bundleCommand = "ram-bundle"
    //
    //   The path to the CLI configuration file. Default is empty.
    // bundleConfig = file(../rn-cli.config.js)
    //
    //   The name of the generated asset file containing your JS bundle
    // bundleAssetName = "MyApplication.android.bundle"
    //
    //   The entry file for bundle generation. Default is 'index.android.js' or 'index.js'
    // entryFile = file("../js/MyApplication.android.js")
    //
    //   A list of extra flags to pass to the 'bundle' commands.
    //   See https://github.com/react-native-community/cli/blob/main/docs/commands.md#bundle
    // extraPackagerArgs = []

    /* Hermes Commands */
    //   The hermes compiler command to run. By default it is 'hermesc'
    // hermesCommand = "$rootDir/my-custom-hermesc/bin/hermesc"
    //
    //   The list of flags to pass to the Hermes compiler. By default is "-O", "-output-source-map"
    // hermesFlags = ["-O", "-output-source-map"]
}
project.ext.react=[
    enableHermes: false
]

apply from: "../../node_modules/react-native-ultimate-config/android/rnuc.gradle"
apply from: file("../../node_modules/react-native-vector-icons/fonts.gradle")
/**
 * Set this to true to Run Proguard on Release builds to minify the Java bytecode.
 */
def enableProguardInReleaseBuilds = false

/**
 * The preferred build flavor of JavaScriptCore (JSC)
 *
 * For example, to use the international variant, you can use:
 * `def jscFlavor = 'org.webkit:android-jsc-intl:+'`
 *
 * The international variant includes ICU i18n library and necessary data
 * allowing to use e.g. `Date.toLocaleString` and `String.localeCompare` that
 * give correct results when using with locales other than en-US. Note that
 * this variant is about 6MiB larger per architecture than default.
 */
def jscFlavor = 'org.webkit:android-jsc:+'

android {
    ndkVersion rootProject.ext.ndkVersion

    compileSdkVersion rootProject.ext.compileSdkVersion

    namespace "com.hphservicesapp"
    defaultConfig {
        applicationId project.config.get("PRODUCT_BUNDLE_IDENTIFIER")
        minSdkVersion rootProject.ext.minSdkVersion
        targetSdkVersion rootProject.ext.targetSdkVersion
        versionCode 1
        versionName "1.0"
    }
    signingConfigs {
        debug {
            storeFile file('debug.keystore')
            storePassword 'android'
            keyAlias 'androiddebugkey'
            keyPassword 'android'
        }
    }
    buildTypes {
        debug {
            signingConfig signingConfigs.debug
        }
        release {
            // Caution! In production, you need to generate your own keystore file.
            // see https://reactnative.dev/docs/signed-apk-android.
            signingConfig signingConfigs.debug
            minifyEnabled enableProguardInReleaseBuilds
            proguardFiles getDefaultProguardFile("proguard-android.txt"), "proguard-rules.pro"
        }
    }
}

dependencies {
    // The version of react-native is set by the React Native Gradle Plugin
    implementation("com.facebook.react:react-android")
    implementation project(':react-native-vector-icons')
    debugImplementation("com.facebook.flipper:flipper:${FLIPPER_VERSION}")
    debugImplementation("com.facebook.flipper:flipper-network-plugin:${FLIPPER_VERSION}") {
        exclude group:'com.squareup.okhttp3', module:'okhttp'
    }

    debugImplementation("com.facebook.flipper:flipper-fresco-plugin:${FLIPPER_VERSION}")
    if (hermesEnabled.toBoolean()) {
        implementation("com.facebook.react:hermes-android")
    } else {
        implementation jscFlavor
    }
}

apply from: file("../../node_modules/@react-native-community/cli-platform-android/native_modules.gradle"); applyNativeModulesAppBuildGradle(project)
```

**Comment by leonardo-coutinho-dev** [comment]
> Fixed in [v6.5.1](https://github.com/lottie-react-native/lottie-react-native/releases/tag/v6.5.1) 🚀

This solved the problem, thank you so much guys! ;)
